storage.go

```diff
@@ -1,44 +1,101 @@
 package main
 
 import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
 	"os"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"go.uber.org/zap"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
-	"go.uber.org/zap"
 )
 
+// TODO: define and implement an interface to allow easier swapping of backends
+
 func newAWSSession(region string) (*session.Session, error) {
 	return session.NewSession(&aws.Config{
 		Region: aws.String(region),
 	})
 }
 
 // taken from https://golangcode.com/uploading-a-file-to-s3/
-func uploadFile(s *session.Session, bucket string, fn string, folder string, platform string, game string) error {
-	file, err := os.Open(fn)
+func uploadFile(logger *zap.Logger, s *session.Session, bucket string, filepath string, folder string, filename string) error {
+	file, err := os.Open(filepath)
 	if err != nil {
 		return err
 	}
 	defer file.Close()
 
-	// TODO: Create value used for `Key` below which incorporates platform/game info
-	// e.g. key = folder/platform-game-fn
+	key := fmt.Sprintf("%s/%s", folder, filename)
+
+	logger.Info(
+		"uploading file to S3 bucket",
+		zap.String("bucket name", bucket),
+		zap.String("filepath", filepath),
+		zap.String("Dest filename", key),
+	)
 
 	uploader := s3manager.NewUploader(s)
 	_, err = uploader.Upload(&s3manager.UploadInput{
 		Bucket: aws.String(bucket),
-		Key:    aws.String(fn),
+		Key:    aws.String(key),
 		Body:   file,
 	})
 
 	return err
 }
 
-func processMediaURLs(urls []string) {}
+func processMediaURLs(config Config, logger *zap.Logger, urls []string, timestamp string, platform string, game string) error {
+	for _, url := range urls {
+		processMediaURL(config, logger, url, timestamp, platform, game)
+	}
+	return nil
+}
+
+func processMediaURL(config Config, logger *zap.Logger, url string, timestamp string, platform string, game string) (string, error) {
```

Comment:
This function will likely include the logic to determine the folder name base on the platform value and defined configuration (or, rather, call a function that will do that).